publisher/internal/analytics/ethereum/operations.go

```diff
@@ -342,24 +326,24 @@ func (rem Removal) Publish(send analytics.Sender, publishTo string, timestamp ti
 		CurrentTokenRatio: rem.CurrentRatio,
 		UpperTokenRatio:   rem.UpperRatio,
 		ValueRemovedUSD:   rem.TotalValue,
+		ValueEarnedUSD:    rem.Token0.Price*rem.Token0Earned.Amount + rem.Token1.Price*rem.Token1Earned.Amount,
 		Pair: [2]types.TokenMessage{
-			{Symbol: rem.Token0.Symbol, Amount: rem.Token0.Amount, Price: rem.Token0.Price},
-			{Symbol: rem.Token1.Symbol, Amount: rem.Token1.Amount, Price: rem.Token1.Price},
+			{Address: rem.Token0.Address, Symbol: rem.Token0.Symbol, Amount: rem.Token0.Amount, Price: rem.Token0.Price},
+			{Address: rem.Token1.Address, Symbol: rem.Token1.Symbol, Amount: rem.Token1.Amount, Price: rem.Token1.Price},
 		},
-		Earned: [2]types.TokensEarnedMessage{
-			{Symbol: rem.Token0.Symbol, Amount: rem.Token0Earned.Amount, TotalValueUSD: rem.Token0.Price * rem.Token0Earned.Amount},
-			{Symbol: rem.Token1.Symbol, Amount: rem.Token1Earned.Amount, TotalValueUSD: rem.Token1.Price * rem.Token1Earned.Amount},
+		Earned: [2]types.TokenMessage{
+			{Symbol: rem.Token0.Symbol, Amount: rem.Token0Earned.Amount},
+			{Symbol: rem.Token1.Symbol, Amount: rem.Token1Earned.Amount},
 		},
-		ValueEarnedUSD: rem.Token0.Price*rem.Token0Earned.Amount + rem.Token1.Price*rem.Token1Earned.Amount,
-		TxHash:         rem.TxHash,
+		TxHash: rem.TxHash,
 	}
 
 	removalJson, err := json.Marshal(&removalMessage)
 	if err != nil {
 		return fmt.Errorf("error marshalling Liquidity Removal object into a json message: %s", err)
 	}
 
-	streamName := strings.ToLower(fmt.Sprintf("%s.%s.%s", publishTo, rem.Token0.Symbol, rem.Token1.Symbol))
+	streamName := strings.ToLower(fmt.Sprintf("%s.%s", publishTo, rem.Address))
 	return send(removalJson, streamName)
```

Comment:
Take a look if this code:
```
	removalJson, err := json.Marshal(&removalMessage)
	if err != nil {
		return fmt.Errorf("error marshalling Liquidity Removal object into a json message: %s", err)
	}

	streamName := strings.ToLower(fmt.Sprintf("%s.%s", publishTo, rem.Address))
```
could be moved inside `send` by using e.g. variable argument list for streamName and accepting `any` instead of bytes for the message body?